### PR TITLE
[CPU] [ARM] Enable Pooling SLT tests on ARM

### PR DESCRIPTION
**Pull request by alvoron**

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/pooling.cpp [hunk 1]
```diff
@@ -271,14 +271,28 @@ void Pooling::getSupportedDescriptors() {
     const auto &childShape = getOutputShapeAtPort(0);
     const size_t inputRank = getInputShapeAtPort(0).getRank();
 
+    inShape = MemoryDescUtils::makeDummyShape(parentShape);
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/pooling.cpp:274]: For static shape we should just use
inShape = parentShape;

File: src/plugins/intel_cpu/src/nodes/pooling.cpp [hunk 2]
```diff
@@ -271,14 +271,28 @@ void Pooling::getSupportedDescriptors() {
...
     // WA: we may specify any layout here (NCHW or NHWC) since both are supported by ACL
-    arm_compute::DataLayout dataLayout = (parentShape.getDims().size() == 5) ? arm_compute::DataLayout::NDHWC : arm_compute::DataLayout::NCHW;
-    arm_compute::TensorInfo srcTensorInfo = arm_compute::TensorInfo(shapeCast(parentShape.getDims()),
+    arm_compute::DataLayout dataLayout = (inShape.getDims().size() == 5) ? arm_compute::DataLayout::NDHWC : arm_compute::DataLayout::NCHW;
+    arm_compute::TensorInfo srcTensorInfo = arm_compute::TensorInfo(shapeCast(inShape.getDims()),
                                                                     1,
                                                                     precisionToAclDataType(inputPrecision),
                                                                     dataLayout);
-    arm_compute::TensorInfo dstTensorInfo = arm_compute::TensorInfo(shapeCast(childShape.getDims()),
+    arm_compute::TensorInfo dstTensorInfo = arm_compute::TensorInfo(shapeCast(MemoryDescUtils::makeDummyShape(childShape).getDims()),
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/pooling.cpp:295]: same here

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/instances/common/pooling.cpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+using namespace CPUTestUtils;
+using namespace ngraph::helpers;
+using namespace ov::test;
+
+namespace CPULayerTestsDefinitions {
+namespace Pooling {
+
+static CPUSpecificParams expectedCpuConfig() {
+#if defined(OPENVINO_ARCH_ARM) || defined(OPENVINO_ARCH_ARM64)
+    return CPUSpecificParams{{}, {}, {"acl"}, "acl"};
```
[the remark this answers is not in the document]
Comment (EgorDuplensky) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/instances/common/pooling.cpp:20]: agreed to split filterCPUInfoForDevice into arch specific implementations in scope of the next commit.

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/instances/common/pooling.cpp [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+                            ngraph::op::RoundingType::CEIL, ngraph::op::PadType::EXPLICIT, false },
+};
+
+INSTANTIATE_TEST_SUITE_P(smoke_MaxPool_CPU_4D, PoolingLayerCPUTest,
+                            ::testing::Combine(
+                            ::testing::ValuesIn(paramsMax4D()),
+                            ::testing::ValuesIn(inputShapes4D()),
+                            ::testing::ValuesIn((inpOutPrecision())),
+                            ::testing::Values(false),
+                            ::testing::ValuesIn(filterCPUInfoForDevice(vecCpuConfigs)),
```
[the remark this answers is not in the document]
Comment (alvoron) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/instances/common/pooling.cpp:73]: fixed

**Review by EgorDuplensky — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**